frontend/src/api/prometheus/__tests__/usePrometheusQueryRange.spec.ts

```diff
@@ -0,0 +1,121 @@
+
+const mockAxios = jest.mocked(axios.post);
+
+describe('usePrometheusQueryRange', () => {
+  const responsePredicate = jest.fn();
+  const queryLang = 'testQuery';
+  const span = 60;
+  const endInMs = 123456;
+  const step = 30;
+  const namespace = 'testNamespace';
+  const active = true;
+
+  it('should fetch data successfully and handle refresh ', async () => {
+    const mockedResponse = { data: { response: mockPrometheusServing({}) } };
+
+    mockAxios.mockResolvedValue(mockedResponse);
+    responsePredicate.mockReturnValue(mockedResponse);
+
+    const renderResult = testHook(usePrometheusQueryRange)(
+      active,
+      '/api/prometheus/serving',
+      queryLang,
+      span,
+      endInMs,
+      step,
+      responsePredicate,
+      namespace,
+    );
+
+    expect(renderResult).hookToStrictEqual([[], false, undefined, expect.any(Function), true]);
+
+    expect(mockAxios).toHaveBeenCalledTimes(1);
+    expect(renderResult).hookToHaveUpdateCount(1);
+
+    await renderResult.waitForNextUpdate();
+
+    expect(renderResult).hookToStrictEqual([
+      mockedResponse,
+      true,
+      undefined,
+      expect.any(Function),
+      false,
+    ]);
+    expect(mockAxios).toHaveBeenCalledTimes(1);
+    expect(renderResult).hookToHaveUpdateCount(2);
+
+    expect(renderResult).hookToBeStable([false, true, false, false, true]);
+
+    await act(() => renderResult.result.current[3]());
+
+    expect(renderResult).hookToStrictEqual([
+      mockedResponse,
+      true,
+      undefined,
+      expect.any(Function),
+      false,
+    ]);
+
+    expect(mockAxios).toHaveBeenCalledTimes(2);
+    expect(renderResult).hookToHaveUpdateCount(3);
+
+    expect(renderResult).hookToBeStable([true, true, false, false, false]);
+  });
+  it('should handle error when active is false', async () => {
+    testHook(usePrometheusQueryRange)(
+      false,
+      '/api/prometheus/serving',
+      queryLang,
+      span,
+      endInMs,
+      step,
+      responsePredicate,
+      namespace,
+    );
+    expect(mockAxios).toHaveBeenCalledTimes(0);
+  });
+  it('should handle error when request fails', async () => {
+    mockAxios.mockRejectedValue(new Error('error'));
+    const renderResult = testHook(usePrometheusQueryRange)(
+      active,
+      '/api/prometheus/serving',
+      queryLang,
+      span,
+      endInMs,
+      step,
+      responsePredicate,
+      namespace,
+    );
+    expect(renderResult).hookToStrictEqual([[], false, undefined, expect.any(Function), true]);
+    expect(mockAxios).toHaveBeenCalledTimes(1);
+    expect(renderResult).hookToHaveUpdateCount(1);
+
+    await renderResult.waitForNextUpdate();
+
+    expect(renderResult).hookToStrictEqual([
+      [],
+      false,
+      new Error('error'),
+      expect.any(Function),
+      false,
+    ]);
+    expect(mockAxios).toHaveBeenCalledTimes(1);
+    expect(renderResult).hookToHaveUpdateCount(2);
+    expect(renderResult).hookToBeStable([true, true, false, false, false]);
```

Comment:
Should assert that refresh function remains stable.
```suggestion
    expect(renderResult).hookToBeStable([true, true, false, true, false]);
```